lib/Compilation.js

```diff
@@ -838,6 +838,7 @@ Compilation.prototype.createChunkAssets = function createChunkAssets() {
 			var usedHash = useChunkHash ? chunkHash : this.fullHash;
 			if(this.cache && this.cache["c" + chunk.id] && this.cache["c" + chunk.id].hash === usedHash) {
 				source = this.cache["c" + chunk.id].source;
+				source.useCache = true;
```

Comment:
Do your mean whether the asset comes from compilation's cache object is judged by `source.source()` content ? 🤔